enhancements/ingress/set_eip_nlb_ingress.md

```diff
@@ -0,0 +1,1319 @@
+
+#### Set EIP using an installer:
+1. Cluster administrator creates `install-config.yaml` and adds the following in it. 
+```yaml
+apiVersion: v1
+baseDomain: example.com
+metadata:
+  name: test-cluster
+platform:
+  aws:
+    region: us-west-2
+    lbType: NLB
+      networkLoadBalancer:
+        eipAllocations:
+        - eipalloc-1111
+        - eipalloc-2222
+        - eipalloc-3333
+        - eipalloc-4444
+        - eipalloc-5555 
+pullSecret: '{"auths": ...}'
+sshKey: ssh-ed25519 AAAA...
+```  
+Here, the administrator needs to know how many public subnets the VPC ID of their cluster has.
+They may check it by clicking the `Load Balancers` tab then under the details click `VPC ID` under VPC, then click the link under `Main route table` and then check the column `Explicit subnet associations` which
+will show you the number of public subnets available for that VPC ID.
+
+Other way to check VPC ID and subnet is go `Your VPCs` under `Virtual Public CLoud` section, then search for the cluster name using the
+command `oc get infrastructure cluster -o jsonpath='{.status.infrastructureName}'`
+
+One can try these scripts present [here](https://github.com/frobware/haproxy-hacks/tree/master/NE-1398) as well.
+
+2. The installer will create the cluster ingress config object as follows:
+```yaml
+ oc edit ingress.config/cluster -o yaml
+ ...
+ apiVersion: config.openshift.io/v1
+ kind: Ingress
+ metadata:
+   creationTimestamp: "2022-08-30T01:03:17Z"
+   generation: 11
+   name: cluster
+   resourceVersion: "60888"
+   uid: de89cc95-449e-4ae5-a837-f4171fe61e36
+ spec:
+   domain: apps.testnlbapi.devcluster.openshift.com
+   loadBalancer:
+     platform:
+       aws:
+         type: NLB
+           networkLoadBalancer:
+             eipAllocations:
+             - eipalloc-1111
+             - eipalloc-2222
+             - eipalloc-3333
+             - eipalloc-4444
+             - eipalloc-5555
+       type: AWS
+```
+3. Ingress Operator will create the default ingress controller with the service type load balancer having the service annotation `service.beta.kubernetes.io/aws-load-balancer-eip-allocations` having values of the field `eipAllocations` 
+   from the Ingress Controller CR.
+
+#### Create an Ingress Controller using Ingress Controller CR and assign the EIP allocation mentioned in the CR to the kubernetes Load Balancer service for the Ingress Controller.
+1. Cluster administrator creates an Ingress Controller:
+```yaml
+ % cat custom-eip-cr.yaml
+ apiVersion: operator.openshift.io/v1
+ kind: IngressController
+ metadata:
+   creationTimestamp: null
+   name: test
+   namespace: openshift-ingress-operator
+ spec:
+   domain: apps1.apps.misalunkeip16.devcluster.openshift.com
+   endpointPublishingStrategy:
+     loadBalancer:
+       scope: External
+       providerParameters:
+         type: AWS
+         aws:
+           type: NLB
+           networkLoadBalancer:
```

Comment:
Seems in PR https://github.com/openshift/cluster-ingress-operator/pull/1034/files the field `networkLoadBalancer` is changed to `nlbParameters` ?
Shall we update the same in install-config.yaml and ingress.config?

Reply:
Yes I will